### DISCUSSION TITLE
Allow `throw redirect('/path')` inside components rendering, just like `loader` and `action`

### DISCUSSION THREAD

**Discussion by remorses** [remix-run/remix#10312] [Proposals]
Currently there is no function to redirect both from `loader` and from a component render function. Inside loader you can throw a redirect, inside a component you have to use an hook.

Remix should intercept errors and checking if they are a Response instance, if they are a redirect response then do a client navigation.

Next.js does something similar with their `redirect()` function

This wouldn't be a breaking change because throwing a Response would simply go to the ErrorBoundary otherwise.

**Comment by akomm** [comment]
Sounds weird, unless you server-render (RSC when implemented) them maybe, but you can use that on the client:
https://remix.run/docs/en/main/hooks/use-navigate
